nodebuilder/header/data_root_tuple_root.go

```diff
@@ -211,19 +204,21 @@ func proveDataRootTuples(tuples []dataRootTuple, height uint64) (*merkle.Proof,
 
 // fetchDataRootTuples takes an end exclusive range of heights and fetches its
 // corresponding data root tuples.
+// end is not included in the range.
 func (s *Service) fetchDataRootTuples(ctx context.Context, start, end uint64) ([]dataRootTuple, error) {
 	tuples := make([]dataRootTuple, 0, end-start)
-	for height := start; height < end; height++ {
-		block, err := s.GetByHeight(ctx, height)
-		if err != nil {
-			return nil, err
-		}
-		if block == nil {
-			return nil, fmt.Errorf("couldn't load block %d", height)
-		}
+	startHeader, err := s.GetByHeight(ctx, start)
+	if err != nil {
+		return nil, err
+	}
+	headerRange, err := s.GetRangeByHeight(ctx, startHeader, end)
+	if err != nil {
+		return nil, err
+	}
+	for _, header := range headerRange {
 		tuples = append(tuples, dataRootTuple{
-			height:   block.Height(),
-			dataRoot: *(*[32]byte)(block.DataHash),
+			height:   header.Height(),
+			dataRoot: *(*[32]byte)(header.DataHash),
```

Comment:
FYI header @ `start` height will not be included in the tuples, is this intentional?

Reply:
Thanks 🙏  
[644c279](https://github.com/celestiaorg/celestia-node/pull/3470/commits/644c2792f91991a83b4d372db019a5561b5b939c)